R/privileges.R

```diff
@@ -0,0 +1,211 @@
+#' Grant
+#'
+#' @export
+#' @param .data an s3 object of class `privilege`
+#' @param ... one of all, select, update, insert, delete
+#' @param cols (character) vector of column names
+#' @examplesIf has_postgres()
+#' library(DBI)
+#' library(RPostgres)
+#' con <- dbConnect(Postgres())
+#' if (!dbExistsTable(con, "passwd")) {
+#'    setup_example_table(con, "passwd")
+#' }
+#'
+#' rls_tbl(con, "passwd") %>%
+#'   grant(update) %>%
+#'   to(jane)
+#'
+#' rls_tbl(con, "passwd") %>%
+#'   grant(update, delete) %>%
+#'   to(jane)
+#'
+#' rls_tbl(con, "passwd") %>%
+#'   grant(update, select, cols = c("real_name", "home_phone")) %>%
+#'   to(jane)
+#' 
```

Comment:
How can I see the results of `grant` in action? My instinct was `rls_column_privileges` but it returns an empty. Also who is `jane` in this context? Do I have to create jane as a user or something first?

Reply:
> How can I see the results of grant in action?

The getting started vignette has use of grant in it. Does that help?

> My instinct was `rls_column_privileges` but it returns an empty

still looking at this...

> Also who is jane in this context? Do I have to create jane as a user or something first?

A role user or role. I'll add more code to make it fully reproducible

Reply:
@seankross 

So one big thing to keep in mind is that the package is currently using behavior like that of `httr2` where the sql is not executed until you use a separate function to run the SQL (`req_perform`). - OR - you can set `auto_pipe(TRUE)` to have policies and privileges code executed at the end of a pipe block

`jane` is a role in the examples. I've fixed examples for `grant` and `revoke` to create and cleanup the role `jane` or `jill` - let me know if these work or not